src/oscar/core/loading.py

```diff
@@ -58,6 +59,17 @@ def get_class(module_label, classname, module_prefix='oscar.apps'):
 
 
 def get_classes(module_label, classnames, module_prefix='oscar.apps'):
+    # Allow projects to supply their own logic for loading classes.
+    # If none is provided then use Oscar's default_class_loader
+    _custom_class_loader = getattr(settings, 'OSCAR_DYNAMIC_CLASS_LOADER', None)
```

Comment:
With a default value specified this function could be a lot simpler, refactor to create `get_class_loader` and then decorate that call with `django.utils.lru_cache` (works with 2 and 3) to cache a reference to the loader.

Reply:
Don't think we need to bother with `lru_cache` if we're just returning a settings attribute. It also makes testing easier ;-).